File: threads-query-app/src/main/java/dao/employee/EmployeeDaoImpl.java
package dao.employee;

import connection.ConnectionClass;
import model.Employee;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class EmployeeDaoImpl implements EmployeeDao{

    @Override
    public Employee findEmployeeById(Integer id) {

        final String SQL = "select employee_id, first_name, last_name, title from employees where employee_id = ?";
        
        try(Connection conn = ConnectionClass.getConnection()){
            try(PreparedStatement st = conn.prepareStatement(SQL)){
                st.setInt(1, id);
                try(ResultSet rs = st.executeQuery()){
                    if(rs.next()){
                        Employee emp = new Employee();
                        emp.setId(id);
                        emp.setFirstName(rs.getString("first_name"));
                        emp.setLastName(rs.getString("last_name"));
                        emp.setTitle(rs.getString("title"));
                        return emp;
                    }
                    else {
                        return null;
                    }
                }
            }

        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;
    }

    @Override
    public List<Employee> getAllEmployees() {
        List<Employee> list = new ArrayList<>();
        final String SQL = "select employee_id, first_name, last_name, title from employees";

        try(Connection conn = ConnectionClass.getConnection()){
            try(PreparedStatement st = conn.prepareStatement(SQL)){
                try(ResultSet rs = st.executeQuery()){
                    while(rs.next()){
                        Employee emp = new Employee();
                        emp.setId(rs.getInt("employee_id"));
                        emp.setFirstName(rs.getString("first_name"));
                        emp.setLastName(rs.getString("last_name"));
                        emp.setTitle(rs.getString("title"));
                        list.add(emp);
                    }
                    return list;
                }
            }
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;
    }
}

File: threads-query-app-kt/src/main/kotlin/dao/employee/EmployeeDaoImpl.kt
package dao.employee

import connection.ConnectionClass
import model.Employee
import java.sql.SQLException

class EmployeeDaoImpl: EmployeeDao {

    override fun findEmployeeById(id: Int): Employee? {

        val sql = "select employee_id, first_name, last_name, title from employees where employee_id = ?"

        try {
            ConnectionClass.getConnection().use { conn ->
                conn.prepareStatement(sql).use { st ->
                    st.setInt(1, id)
                    st.executeQuery().use { rs ->
                        return if (rs.next()) {
                            val emp = Employee (
                                id = id,
                                firstName = rs.getString("first_name"),
                                lastName = rs.getString("last_name"),
                                title = rs.getString("title")
                            )
                            emp
                        }
                        else null
                    }
                }
            }
        }
        catch (e: SQLException) {
            e.printStackTrace()
        }
        return null
    }

    override fun getAllEmployees(): List<Employee>? {
        val list: MutableList<Employee> = ArrayList()
        val sql = "select employee_id, first_name, last_name, title from employees"

        try{
            ConnectionClass.getConnection().use { conn ->
                conn.prepareStatement(sql).use { st ->
                    st.executeQuery().use { rs ->
                        while (rs.next()) {
                            list.add(
                                Employee(
                                    id = rs.getInt("employee_id"),
                                    firstName = rs.getString("first_name"),
                                    lastName = rs.getString("last_name"),
                                    title = rs.getString("title")
                                )
                            )
                        }
                        return list
                    }
                }
            }
        }
        catch (e: SQLException) {
            e.printStackTrace()
        }
        return null
    }
}